opensquirrel/passes/decomposer/general_decomposer.py

```diff
@@ -18,8 +20,10 @@ def decompose(self, gate: Gate) -> list[Gate]:
 def check_gate_replacement(gate: Gate, replacement_gates: Iterable[Gate]) -> None:
     gate_qubit_indices = [q.index for q in gate.get_qubit_operands()]
     replacement_gates_qubit_indices = set()
+    replaced_matrix = get_circuit_matrix(get_reindexed_circuit([gate], gate_qubit_indices))
+    norm_matrix = np.abs(replaced_matrix)
 
-    if gate.is_identity():
+    if np.allclose(norm_matrix, np.eye(norm_matrix.shape[0]), atol=ATOL):
```

Comment:
Can we make this two lines into a function? Beware name `is_normalized` may be incorrect. E.g.:

```
def is_normalized(matrix: Matrix):
    norm_matrix = np.abs(matrix)
    return np.allclose(norm_matrix, np.eye(norm_matrix.shape[0]), atol=ATOL):

if matrix_is_normalized(replaced_matrix):
    return
```